gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,286 @@
+            args: ["{that}"]
+        },
+        getSystemSettingsCapture: {
+            funcName: "gpii.flowManager.getSystemSettingsCapture",
+            args: ["{that}.events.onCaptureSettingsForCurrentDevice", "{arguments}.0"] // Options
+        },
+        captureSystemSettings: {
+            funcName: "gpii.flowManager.captureSystemSettings",
+            args: ["{lifecycleManager}.invokeSettingsHandlerGet", "{lifecycleManager}.variableResolver",
+                "{arguments}.0", "{arguments}.1"]
+        }
+    }
+});
+
+gpii.flowManager.getInstalledSolutions = function (that) {
+    return fluid.promise.fireTransformEvent(that.events.onSolutionsForCurrentDevice);
+};
+
+/**
+ * This main API entry point for capturing settings from a system or computer. This captures
+ * the actual settings on the device, so it assumed to be running in a local untrusted flow
+ * manager.
+ *
+ * @param {Event} onCaptureSettingsForCurrentDevice - The transforming promise chain
+ * @param {Object} options - Options for this chain.
+ * @param {Array} options.solutionsList - An array of solution IDs to filter by when
+ * retreiving settings. If this option is not included, all available settings will be
+ * returned. ex: `["com.microsoft.windows.mouseSettings", "com.freedomscientific.jaws"]`.
+ * @return {Promise} A promise resolved with the payload of captured system settings.
+ */
+gpii.flowManager.getSystemSettingsCapture = function (onCaptureSettingsForCurrentDevice, options) {
+    return fluid.promise.fireTransformEvent(onCaptureSettingsForCurrentDevice,
+        {}, options);
+};
+
+/**
+ * This function invokers a settings handler to fetch settings, but wraps the processing and reports
+ * back any failures in an `isError: true` json block.
+ *
+ * @param {Function} invokeSettingsHandlerGet - The get function from the appropriate settings handler.
+ * @param {String} solutionID - The dotted solution id, ex `com.freedomscientific.jaws`
+ * @param {Object} handlerSpec - TODO, exactly how much of the solutions entry block is this?
+ * @return {Object} The returned settings, or an error block with debugging information.
+ */
+gpii.flowManager.capture.safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+    var promiseTogo = fluid.promise();
+    try {
+        invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+            promiseTogo.resolve(data);
+        }, function (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "Settings Handler Promise Rejection",
+                err: err
+            });
+        });
+    }
+    catch (err) {
+        promiseTogo.resolve({
+            solutionId: solutionID,
+            handlerSpec: handlerSpec,
+            isError: true,
+            message: "Unexpected Error in operating settings handler.",
```

Comment:
Mention the capture workflow in both of these messages. You may find the utility kettle.upgradeError https://github.com/fluid-project/kettle/blob/master/lib/KettleUtils.js#L82 helpful in capturing details of an original error whilst composing further detail

Reply:
I updated these strings to mention the capture workflow, but kept my structure, as I'd some of the data to stay separate for debugging and blackbox investigation if necessary, I don't really want my message and the error concatenated, and would like the other bits to stay in json form.